docs/usage/data-modeling.md

```diff
@@ -0,0 +1,150 @@
+  - `balance = credits - debits` for accounts representing the database operator's liabilities.
+    - Known as a "credit balance".
+    - Account limits use
+      [`flags.debits_must_not_exceed_credits`](../reference/accounts.md#flagsdebits_must_not_exceed_credits).
+  - A transfer "from" account `A` "to" account `B` credits account `A` and debits account `B`.
+
+### Example
+
+For example, if TigerBeetle is operated by a bank, with customers Alice and Bob, its
+ledger might look something like this:
+
+| Account Owner | Debits Posted | Credits Posted | Flags                            |
+| :------------ | ------------: | -------------: | :------------------------------- |
+| Bank          |            30 |              0 | `credits_must_not_exceed_debits` |
+| Alice         |             0 |             20 | `debits_must_not_exceed_credits` |
+| Bob           |             0 |             10 | `debits_must_not_exceed_credits` |
+
+- The bank has a total of $30 in assets.
+- Alice and Bob have deposited money ($20 and $10 respectively) in the bank — from the bank's
+  perspective this is a liability.
+- Alice and Bob cannot "overdraw" their account — that is, their balance will never be negative.
+
+## `user_data`
+
+`user_data` is the most flexible field in the schema (for both
+[accounts](../reference/accounts.md#user_data) and [transfers](../reference/transfers.md#user_data)).
+`user_data`'s contents are arbitrary, interpreted only by the application.
+
+`user_data` is indexed for efficient point and range queries.
+
+Example uses:
+
+- Set `user_data` to a "foreign key" — that is, an identifier of a corresponding object within
+  another database.
+- Set `user_data` to a group identifier for objects that will be queried together.
+- Set `user_data` to a transfer or account `id`.
+  (TODO: Can we use this for join queries via the query API, or must the application implement them?)
+
+## `id`
+
+The primary purpose of an `id` (for both [accounts](../reference/accounts.md#id) and
+[transfers](../reference/transfers.md#id)) is to serve as an "idempotency key".
+The database uses ids to avoid repeating duplicate events.
+
+[Randomly-generated identifiers](#random-identifier) are recommended for most applications.
+
+When selecting an `id` scheme:
+
+- Idempotency is particularly important (and difficult) in the context of
+  [application crash recovery](./integration.md#consistency-with-foreign-databases).
+- Be careful to [avoid `id` collisions](https://en.wikipedia.org/wiki/Birthday_problem).
+- An account and a transfer may share the same `id` — they belong to different "namespaces".
+- Avoid requiring a central oracle to generate each unique `id` (e.g. an auto-increment field in SQL).
+  A central oracle may become a performance bottleneck when creating accounts/transfers.
+- `id` is mostly accessed by point queries, but it is indexed for efficient range queries as well.
```

Comment:
If ids are random the bit about efficient range queries doesn't make a ton of sense but we aren't requiring random ids I guess

Reply:
Yeah, the range query part is about non-random-ids.